prompt_poet/template.py

```diff
@@ -34,15 +38,37 @@ def __init__(
             raise ValueError(
                 f"Cannot provide both {raw_template=} and {template_path=}."
             )
+        if template_path:
+            if logger:
+                logger.warning(
+                    "`template_path` is deprecated and will be removed in a future release. Use `template_loader` instead.")
+            else:
+                print(
+                    "Warning: `template_path` is deprecated and will be removed in a future release. Use `template_loader` instead.")
+
+        if package_name:
+            if logger:
+                logger.warning(
+                    "`package_name` is deprecated and will be removed in a future release. Use `template_loader` instead.")
+            else:
+                print(
+                    "Warning: `package_name` is deprecated and will be removed in a future release. Use `template_loader` instead.")
```

Comment:
you should restructure this code so that you can use self.logger to handle the case where logger input param is not provided, rather than writing the warning string twice once in a print and once as a logger warning.

alternatively altogether you could consider using `warnings` module https://docs.python.org/3/library/warnings.html

Reply:
updated